rasa/core/actions/action.py

```diff
@@ -591,7 +591,103 @@ def action_response_format_spec() -> Dict[Text, Any]:
                     "type": "array",
                     "items": {
                         "type": "object",
-                        "properties": {"event": {"type": "string"}},
+                        "properties": {"event": {"type": "string"}, "timestamp": {}},
+                        "oneOf": [
+                            {
+                                "properties": {
+                                    "event": {"const": "user"},
+                                    "text": {"type": "string"},
+                                    "input_channel": {},
+                                    "message_id": {"type": "string"},
+                                    "parse_data": {
+                                        "type": "object",
+                                        "properties": {
+                                            "text": {"type": "string"},
+                                            "intent": {
+                                                "type": "object",
+                                                "properties": {
+                                                    "name": {"type": "string"},
+                                                    "confidence": {"type": "number"},
+                                                },
+                                            },
+                                            "entities": {
+                                                "type": "array",
+                                                "items": {
+                                                    "type": "object",
+                                                    "properties": {
+                                                        "start": {"type": "integer"},
+                                                        "end": {"type": "integer"},
+                                                        "name": {"type": "string"},
+                                                        "confidence": {
+                                                            "type": "number"
+                                                        },
+                                                        "extractor": {"type": "string"},
+                                                        "value": {"type": "object"},
+                                                    },
+                                                },
+                                            },
+                                        },
+                                    },
+                                }
+                            },
+                            {
+                                "properties": {
+                                    "event": {"const": "action"},
+                                    "policy": {},
+                                    "confidence": {},
```

Comment:
optional number?

Reply:
Actually type `Any`, if there's an equivalent for Optional type in JSON schema, I wasn't able to find it.